Client/app/src/main/java/com/acme/snapgreen/ui/dashboard/StatsActivity.kt

```diff
@@ -170,14 +170,16 @@ class StatsActivity : AppCompatActivity() {
         bars.setDrawValues(false)
 
         val lineData = ArrayList<Entry>()
-        lineData.add(Entry(0f, 100f))
-        lineData.add(Entry(1f, 90f))
-        lineData.add(Entry(2f, 110f))
-        lineData.add(Entry(3f, 84f))
-        lineData.add(Entry(4f, 75f))
-        lineData.add(Entry(5f, 99f))
-        lineData.add(Entry(6f, 100f))
-        val lines = LineDataSet(lineData, "Average Data")
+        var average: Float
+        var waterCount = 0f
+        var dayCount2 = 0f
+        for (stat in stats) {
```

Comment:
I would lift the daily data / weekly data setting into different functions (you can do this very easily by selecting the code and pulling it out into a separate function) and comment on each one individually to make the flow clearer. Other than that, lgtm!